pygen/pygen_src/riscv_directed_instr_lib.py

```diff
@@ -11,34 +11,99 @@
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 """
 
+import random
 import vsc
 from enum import IntEnum, auto
 from pygen_src.riscv_instr_stream import riscv_rand_instr_stream
-from pygen_src.isa.riscv_instr import riscv_instr_ins
+from pygen_src.isa.riscv_instr import riscv_instr, riscv_instr_ins
 from pygen_src.riscv_instr_gen_config import cfg
-from pygen_src.riscv_instr_pkg import riscv_reg_t, riscv_pseudo_instr_name_t
+from pygen_src.riscv_instr_pkg import riscv_reg_t, riscv_pseudo_instr_name_t, riscv_instr_name_t
 from pygen_src.target.rv32i import riscv_core_setting as rcs
 from pygen_src.riscv_pseudo_instr import riscv_pseudo_instr
 
 
 class riscv_directed_instr_stream(riscv_rand_instr_stream):
 
+    label = ""
+
     def __init__(self):
         super().__init__()
         self.name = ""
-        self.label = ""
 
     def post_randomize(self):
         for i in range(len(self.instr_list)):
             self.instr_list[i].has_label = 0
             self.instr_list[i].atomic = 1
         self.instr_list[0].comment = "Start %0s" % (self.name)
         self.instr_list[-1].comment = "End %0s" % (self.name)
-        if self.label != "":
-            self.instr_list[0].label = self.label
+        if riscv_directed_instr_stream.label != "":
+            self.instr_list[0].label = riscv_directed_instr_stream.label
             self.instr_list[0].has_label = 1
 
 
+@vsc.randobj
+class riscv_jal_instr(riscv_rand_instr_stream):
```

Comment:
just to confirm, when you run tests with this stream enabled, are they able to compile/simulate with ISS successfully?

Reply:
Yes, Checked with both the ISS (Spike and OVPSim).

Reply:
great!